test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go

```diff
@@ -605,15 +604,38 @@ var _ = f.CLICoreDescribe("[Tests:E2E][Feature:Plugin-Sync-TMC-lifecycle]", func
 			Expect(f.CheckAllPluginsExists(installedPluginsListTMC, pluginsToGenerateMockResponse)).Should(BeTrue(), pluginsInstalledAndMockedShouldBeSame)
 		})
 
-		// Test case: h. validate plugin list consistancy, it should sort by context name always
+		// Test case: h. validate plugin list consistency, it should sort by context name always
 		It("Test case: h: list plugins and validate plugins being installed after context being created", func() {
+			// IsSortedByName checks if the array of objects is sorted by name
+			IsSortedByName := func(arr []*f.PluginInfo) bool {
+				n := len(arr)
+
+				// Iterate through the array to check if it is sorted
+				for i := 1; i < n; i++ {
+
+					// Compare the names of adjacent objects
+					if arr[i-1].Name > arr[i].Name {
+						return false
+					}
+				}
+
+				// If the loop completes without returning false, the array is sorted
+				return true
+			}
 			// check multiple times, the order should be consistent
 			for j := 0; j < 5; j++ {
 				installedPlugins, err := tf.PluginCmd.ListInstalledPlugins()
 				Expect(err).To(BeNil(), noErrorForPluginList)
 				Expect(totalInstalledPlugins).Should(Equal(len(installedPlugins)), "total installed plugins count should be equal to plugins installed for both contexts")
-				sort.Strings(contexts)
-				Expect(f.ValidateInstalledPluginsOrder(contexts, installedPlugins)).To(BeTrue())
+
+				// Filter plugins without context
+				var installedPluginsWithContext []*f.PluginInfo
+				for _, plugin := range installedPlugins {
+					if plugin.Context != "" {
+						installedPluginsWithContext = append(installedPluginsWithContext, plugin)
```

Comment:
this array will contain the plugins from both contexts right?  k8s and tmc.
Those are not sorted alphabetically, but by context, then alphabetically.  I don't understand how the test is passing (although I don't know what the plugins are named, so maybe we are lucky and they are all alphabetical anyway?)

Do you understand what is happening?

Reply:
@marckhouzam  Originally I thought the same but it looks like the plugin list data is sorted by name only. 
This is the response of plugin name and plugin context
```
[i] Executing command: tanzu plugin list -o json
[account apply audit cluster feature kubernetes-release]
[plugin-sync-tmc-FVYj plugin-sync-tmc-FVYj plugin-sync-tmc-FVYj plugin-sync-k8s-UZCH plugin-sync-k8s-UZCH plugin-sync-k8s-UZCH]
```

Reply:
Oh wow, you are right!  Sorry about that.